lib/rack/reverse_proxy.rb

```diff
@@ -129,5 +129,12 @@ def reverse_proxy(matcher, url=nil, opts={})
       raise GenericProxyURI.new(url) if matcher.is_a?(String) && url.is_a?(String) && URI(url).class == URI::Generic
       @matchers << ReverseProxyMatcher.new(matcher,url,opts)
     end
+
+    def format_headers(headers)
+      headers.each_with_object({}).each do |(key, val), acc|
+        formated_key = key.split('-').map(&:capitalize).join('-')
```

Comment:
This code is not well tested. I have added the commit on top: https://github.com/waterlink/rack-reverse-proxy/commit/ba4b9eb00ac60175f57a58ae0fc743eb94b7a178